torch/testing/_internal/common_methods_invocations.py

```diff
@@ -187,6 +195,15 @@ def default_test_dtypes(self, device_type):
 S = 5
 
 
+def sample_inputs_unary(self, device, dtype, requires_grad):
```

Comment:
@Lilyjjo Is it intentional that sample_inputs_unary(self, device, dtype, requires_grad) is not UnaryUfuncInfo class method ?

Reply:
@vfdev-5 Yeah it's a little wonky, but it allows individual ops to control their sample_inputs. I'm a little rough on my python skills, so if you have a suggestion on how to refactor the functionality in a more pythonic way we could def change it

Reply:
@Lilyjjo I see. Maybe, a minimal fix can be like 
```python
def sample_inputs_unary(op_info, device, dtype, requires_grad):
     ...
```
or we can specify a base class to inherit from:
```python
class BaseSampleInputsGenerator

    def __init__(self, *args, **kwargs):
        pass

    def __call__(self, op_info, device, dtype, requires_grad, **kwargs):
        return []

class UnarySampleInputsGenerator(BaseSampleInputsGenerator):

    def __call__(self, op_info, device, dtype, requires_grad, **kwargs):        
        low, high = op_info.domain
        low = low if low is None else low + op_info._domain_eps
        high = high if high is None else high - op_info._domain_eps
        return (SampleInput(make_tensor((L,), device, dtype,
                                          low=low, high=high,
                                          requires_grad=requires_grad)),)
```
In this case we can control function's signature once assigned inside OpInfo
```python
class OpInfo:
    def __init__(self, ..., sample_inputs_gen=None):
         if sample_inputs_gen is not None:
             if not isinstance(sample_inputs_gen, BaseSampleInputsGenerator):
                 raise TypeError("...")

```

Reply:
Changing the name from `self` to `op_info` seems clear, simple, and flexible. We should probably do that as part of another OpInfo PR (to avoid going through the merge process for such a small change).